src/Classes/ConfigTab.lua

```diff
@@ -870,3 +940,63 @@ function ConfigTabClass:RestoreUndoState(state)
 	self:UpdateControls()
 	self:BuildModList()
 end
+
+function ConfigTabClass:OpenConfigSetManagePopup()
+	main:OpenPopup(370, 290, "Manage Config Sets", {
+		new("ConfigSetListControl", nil, 0, 50, 350, 200, self),
+		new("ButtonControl", nil, 0, 260, 90, 20, "Done", function()
+			main:ClosePopup()
+		end),
+	})
+end
+
+-- Creates a new config set
+function ConfigTabClass:NewConfigSet(configSetId, title)
+	local configSet = { id = configSetId, title = title, input = { }, placeholder = { } }
+	if not configSetId then
+		configSet.id = 1
+		while self.configSets[configSet.id] do
+			configSet.id = configSet.id + 1
+		end
+	end
+	-- there are default values for input and placeholder that every new config set needs to have
+	for _, varData in ipairs(varList) do
+		if varData.var then
+			configSet.input[varData.var] = varData.defaultState
+			configSet.placeholder[varData.var] = varData.defaultPlaceholderState
+			if varData.defaultIndex then
+				configSet.input[varData.var] = varData.list[varData.defaultIndex].val
+			end
+		end
+	end
+	self.configSets[configSet.id] = configSet
+	return configSet
+end
+
+-- Changes the active config set
+function ConfigTabClass:SetActiveConfigSet(configSetId, init)
+	-- Initialize config sets if needed
+	if not self.configSetOrderList[1] then
+		self.configSetOrderList[1] = 1
+		self:NewConfigSet(1)
+	end
+
+	if not configSetId then
+		configSetId = self.activeConfigSetId
+	end
+
+	if not self.configSets[configSetId] then
+		configSetId = self.configSetOrderList[1]
+	end
+
+	if not init then
+		self.input = {}
```

Comment:
Do these need to be set to empty tables first?

Reply:
Can't remember why I did that now, not seeing any reason for it. Maybe some values were not clearing when changing configSets but again, not seeing that now. Getting rid of it.